device/tt_device/blackhole_tt_device.cpp

```diff
@@ -13,20 +14,70 @@ BlackholeTTDevice::BlackholeTTDevice(std::unique_ptr<PCIDevice> pci_device) :
     TTDevice(std::move(pci_device), std::make_unique<blackhole_implementation>()) {}
 
 BlackholeTTDevice::~BlackholeTTDevice() {
+    // Turn off iATU for the regions we programmed.  This won't happen if the
+    // application crashes -- this is a good example of why userspace should not
+    // be touching this hardware resource directly -- but it's a good idea to
+    // clean up after ourselves.
```

Comment:
Since we're arguing already around user mode vs kernel mode... If these regions are viewed as resources, then they are administered and when they're administered you can setup/clear them. It is exactly the same thing if this code is in KMD or in UMD with synchronization.

If it is in KMD, and user using it crashes, then KMD can figure out that process doesn't exist, and it can clear the regions and mark regions available for administering them to another user.

If it is in UMD, you try to take the mutex, you get it only if the other process doesn't hold it (it won't hold it if it crashes), then when you know you have the resource you clear it up and use it.

No action for PR